pkg/admin/controller.go

```diff
@@ -44,14 +44,31 @@ func NewController(
 	}
 }
 
-// HandleGetApps handles GET requests
-func (ctrl *Controller) HandleGetApps(w http.ResponseWriter, _ *http.Request) {
-	appNames := ctrl.adminService.GetApps()
+// HandleGetAppNames handles GET requests
+func (ctrl *Controller) HandleGetAppNames(w http.ResponseWriter, r *http.Request) {
+	appNames := ctrl.adminService.GetAppNames(r.Context())
 
 	w.WriteHeader(http.StatusOK)
 	ctrl.writeResponseJSON(w, appNames)
 }
 
+type AppInfo struct {
+	Name string `json:"name"`
+}
+
+func (ctrl *Controller) HandleGetApps(w http.ResponseWriter, r *http.Request) {
+	apps := ctrl.adminService.GetApps(r.Context())
+	res := make([]AppInfo, 0, len(apps.Apps))
+	for _, app := range apps.Apps {
+		it := AppInfo{
+			Name: app.Name,
+		}
+		res = append(res, it)
+	}
+	w.WriteHeader(http.StatusOK)
+	ctrl.writeResponseJSON(w, res)
+}
```

Comment:
Please consider refactoring app handlers to `pkg/server` package due to:
1. Consistency with the existing HTTP API implementation.
2. Proper error handling – HTTP API methods use [a special helper](https://github.com/pyroscope-io/pyroscope/blob/c6486ffbee94f0a7277aa15f40b9000a796c196b/pkg/server/render.go#L81).
3. Independence of admin controller on the initialisation: now the main controller depends on the admin one.